connectivedx/fuzzy-chainsaw
# Enhancement/selector merge

Repeated style blocks was something we use to consolidate down in our productions builds long ago. I'm not sure why we missed this in FC, but I've found a suitable POSTCSS plugin to accomplish this for us again.

## Description
Adds new postcss-merge-selector to package.json
Adds new postcss-merge-selector to postcss-plugins.js production hook.

## Motivation and Context
When productions builds are ran, the build will consolidate down all selectors who have repeated style blocks, instead of repeating said style block X times within bundled sheet.

This allows us to never need to worry about trying to put repeated style blocks in a single location, but instead know that no matter where the styles live they will be consolidate down.

The moment a repeated style block changes, it will break away from the consolidated selector and be considered unique again.

## How Has This Been Tested?
- Add repeated styles with different selectors across two different element css files.
- Run npm or yarn install
- Run npm or yarn production
- After built, check the dist/assets css bundle and locate repeated style selectors. 
- Notice that their selectors are now concatenated together.

## Screenshots (if appropriate):

## Types of changes


## Checklist:
